libpromises/addr_lib.c

```diff
@@ -42,6 +42,12 @@ int FuzzySetMatch(const char *s1, const char *s2)
     short isCIDR = false, isrange = false, isv6 = false, isv4 = false;
     char address[CF_ADDRSIZE];
 
+    if (strlen(s1) >= CF_MAX_IP_LEN)
+    {
+        Log(LOG_LEVEL_ERR, "IP set string '%s' is too large to parse (>=%d)", s1, CF_MAX_IP_LEN);
+        return -1;
```

Comment:
With a deeper look, this seems like it will cause error messages printing many times on each connection, if the ACL has long hostnames included.

Where is the possible overflow? Is it later in this function? If so, just limit that one.